src/item.cpp

```diff
@@ -1523,6 +1523,11 @@ std::string item::info( bool showtext, std::vector<iteminfo> &info ) const
                 }
             }
         }
+        
+        if( is_gun() && has_flag( "FIRE_TWOHAND" ) ) {
+            info.push_back( iteminfo( "DESCRIPTION",
+                                      _( "* This weapon needs <info>two free hands</info> to fire." ) ) );
```

Comment:
It would be better if this was a part of interface, so that you would know if you have two free hands at any time. Worn hand-restricting armor is not too permanent to consider this a bad or good thing because of it, in my opinion.